core/dbsecurity/auth/src/authEvents.cpp

```diff
@@ -0,0 +1,134 @@
+  DB_SECURITY_EVENTID eventID,
+  const char * eventText,
+  logLevel severity)
+{
+  AuthEvent authEvent(eventID,
+                      AuthEvent::formatEventText(eventText),
+                      severity);
+  authEvents.push_back(authEvent);
+}
+
+// ****************************************************************************
+// function authInitEventLog()
+//
+// This function create a new log in $MY_SQROOT/logs directory with the 
+// following name dbsecurity_<host>_<pid>.log
+//
+// It is called for standalone executables (e.g. ldapcheck) to log issues
+// When users are authenticated during client authentication (mxosrvr), it
+//   is assumed that the event log has already been initialized.
+//
+// ****************************************************************************
+void authInitEventLog()
+{
+  // Log4cxx logging
+  int my_nid = 1;
+
+  // get my pid
+  my_nid = getpid();
+  char my_hostname[HOST_NAME_MAX+1];
+  Int32  result;
+
+  // who am I?
+  if (gethostname(my_hostname,HOST_NAME_MAX) != 0)
+    strcpy(my_hostname, "unknown");
+
+  int log_name_suffix_len = strlen(my_hostname) + 32;
+  char log_name_suffix[log_name_suffix_len];
+  snprintf( log_name_suffix, log_name_suffix_len, "_%s_%d.log", my_hostname, my_nid );
+  CommonLogger::instance().initLog4cxx("log4cxx.trafodion.auth.config",log_name_suffix);
+}
+
+// ****************************************************************************
+// method: AuthEvent::formatEventText
+//
+// This method formats event text into a message that can be added to the 
+//   cxx log.
+//
+//  eventText - the text to be logged
+// ****************************************************************************
+std::string AuthEvent::formatEventText( const char * eventText )
+{
+  // Format the timestamp
+   char tbuff[24] = {0};
+   struct tm *stm;
+
+   time_t now = time(0);
+   stm = gmtime(&now);
+   strftime(tbuff, sizeof(tbuff), "%Y-%m-%d %H:%M:%S %Z", stm);
+
+   // Format the event message, with the timestamp at the beginning.
+   char eventMessage[MAX_EVENT_MSG_SIZE];
+   snprintf(eventMessage, MAX_EVENT_MSG_SIZE, "%s (pid=%d) %s", tbuff, getpid(), eventText);
+
+   std::string newMessage(eventMessage);
+   return newMessage;
+}
+
+// ****************************************************************************
+// method: AuthEvent::logAuthEvent
+//
+// This method writes the AuthEvent to the cxx log
+//
+// ****************************************************************************
+void AuthEvent::logAuthEvent()
+{
+    int my_nid = 0; //gethostid()? 
+    int my_pid = getpid();
+    int my_cpu = 0;
+  
+    // Log4cxx logging
+    char buf[MAX_EVENT_MSG_SIZE];
+    snprintf(buf, MAX_EVENT_MSG_SIZE, "Node Number: %u, CPU: %u, PIN: %u ,,,, Message: %s", 
+            my_nid, my_cpu, my_pid, eventText_.c_str());
+    
+    // strip off final new line before logging
+    int32_t lastPos = strlen(buf) -1;
+    if (buf [lastPos] == '\n')
+      buf[lastPos] = '.';
```

Comment:
Do you want a period here or a null character? If the message already ends with a period, you'll get two here.